modules/livestore/instance.go

```diff
@@ -125,12 +135,64 @@ func newInstance(instanceID string, cfg Config, wal *wal.WAL, overrides override
 	return i, nil
 }
 
-func (i *instance) pushBytes(ts time.Time, req *tempopb.PushBytesRequest) {
+func (i *instance) backpressure(ctx context.Context) bool {
```

Comment:
Bikeshedding, thoughts on making the name something like needsBackpressure? Overall not to hung up on the naming though.